### ISSUE TITLE
Can't run linter exportloopref

### ISSUE THREAD

**Issue by cbandy** [golangci/golangci-lint#1515]
It doesn't seem to be possible to run `exportloopref` by itself, even on the simplest of packages. For example, this single file that imports `fmt` can't be analyzed:

```go
package main

import "fmt"

func main() {
	fmt.Println("hello")
}
```

```console
$ golangci-lint run --no-config --disable-all --enable exportloopref
WARN [runner] Can't run linter exportloopref: inspect: analysis skipped: errors in package: [/Users/cbandy/tmp/go/main.go:3:8: could not import fmt (Config.Importer.Import(fmt) returned nil but no error)]
ERRO Running error: inspect: analysis skipped: errors in package: [/Users/cbandy/tmp/go/main.go:3:8: could not import fmt (Config.Importer.Import(fmt) returned nil but no error)]
```

```console
$ golangci-lint run --no-config --disable-all --enable exportloopref --verbose
INFO [lintersdb] Active 1 linters: [exportloopref]
INFO [loader] Go packages loading at mode 7 (compiled_files|files|name) took 322.614921ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 1.58944ms
INFO [linters context/goanalysis] analyzers took 66.964µs with top 10 stages: exportloopref: 45.094µs, inspect: 21.87µs
WARN [runner] Can't run linter exportloopref: inspect: analysis skipped: errors in package: [/Users/cbandy/tmp/go/main.go:3:8: could not import fmt (Config.Importer.Import(fmt) returned nil but no error)]
INFO [runner] processing took 8.453µs with stages: nolint: 2.748µs, max_same_issues: 2.44µs, cgo: 431ns, skip_dirs: 409ns, path_prettifier: 350ns, autogenerated_exclude: 306ns, path_prefixer: 291ns, filename_unadjuster: 260ns, max_from_linter: 203ns, skip_files: 148ns, uniq_by_line: 137ns, diff: 137ns, identifier_marker: 131ns, sort_results: 73ns, max_per_file_from_linter: 68ns, source_code: 67ns, exclude: 67ns, path_shortener: 66ns, exclude-rules: 61ns, severity-rules: 60ns
INFO [runner] linters took 2.534605ms with stages: exportloopref: 2.482917ms
ERRO Running error: inspect: analysis skipped: errors in package: [/Users/cbandy/tmp/go/main.go:3:8: could not import fmt (Config.Importer.Import(fmt) returned nil but no error)]
INFO Memory: 6 samples, avg is 71.6MB, max is 71.6MB
INFO Execution took 405.393236ms
```

I've tried the latest release and built from `@master`.

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.32.2 built from 52d26a3 on 2020-11-03T01:15:38Z
```
and
```
golangci-lint has version v1.32.3-0.20201117214611-c68692e99f34 built from (unknown, mod sum: "h1:TwcnmNXfBsx0UdOb3BaF0B2PYYdtEV11+Ad1qP1xMD4=") on (unknown)
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.15.3 darwin/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/cbandy/Library/Caches/go-build"
GOENV="/Users/cbandy/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/cbandy/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/cbandy/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/Users/cbandy/.local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/Users/cbandy/.local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/cbandy/tmp/go/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/bw/_dxdzsmx6h1ffcgp6rvfh55c0000gn/T/go-build901332976=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/cbandy/tmp/go /Users/cbandy/tmp /Users/cbandy /Users /]
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck]
INFO [loader] Go packages loading at mode 575 (deps|exports_file|files|imports|name|compiled_files|types_sizes) took 289.052724ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 191.856µs
INFO [linters context/goanalysis] analyzers took 974.81474ms with top 10 stages: buildir: 705.474977ms, fact_purity: 86.725757ms, ctrlflow: 55.909102ms, inspect: 46.534415ms, printf: 44.046748ms, fact_deprecated: 34.00492ms, S1030: 131.503µs, ineffassign: 100.069µs, SA4021: 93.871µs, isgenerated: 89.483µs
INFO [linters context/goanalysis] analyzers took 2.766319ms with top 10 stages: buildir: 2.605689ms, U1000: 160.63µs
INFO [runner] processing took 7.439µs with stages: max_same_issues: 2.608µs, nolint: 1.929µs, skip_dirs: 374ns, cgo: 275ns, max_from_linter: 271ns, path_prettifier: 270ns, skip_files: 218ns, filename_unadjuster: 179ns, path_prefixer: 172ns, autogenerated_exclude: 168ns, uniq_by_line: 151ns, diff: 146ns, identifier_marker: 145ns, path_shortener: 80ns, source_code: 80ns, max_per_file_from_linter: 78ns, sort_results: 77ns, exclude: 75ns, exclude-rules: 73ns, severity-rules: 70ns
INFO [runner] linters took 2.138641413s with stages: goanalysis_metalinter: 2.101323097s, unused: 37.254885ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 26 samples, avg is 150.1MB, max is 207.4MB
INFO Execution took 2.441623457s
```

</details>

**Comment by cbandy** [comment]
Enabling any of the other default linters seems to work around this. For example,

```console
$ golangci-lint run --no-config --disable-all --enable exportloopref --enable govet --verbose
INFO [lintersdb] Active 2 linters: [exportloopref govet]
INFO [loader] Go packages loading at mode 575 (files|exports_file|deps|imports|name|types_sizes|compiled_files) took 210.506084ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 184.219µs
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [runner] processing took 6.243µs with stages: max_same_issues: 1.913µs, nolint: 1.754µs, skip_dirs: 398ns, cgo: 236ns, filename_unadjuster: 233ns, max_from_linter: 226ns, skip_files: 216ns, path_prefixer: 146ns, path_prettifier: 141ns, identifier_marker: 136ns, uniq_by_line: 135ns, diff: 134ns, autogenerated_exclude: 134ns, source_code: 66ns, max_per_file_from_linter: 65ns, sort_results: 63ns, path_shortener: 63ns, exclude-rules: 63ns, exclude: 61ns, severity-rules: 60ns
INFO [runner] linters took 25.639343ms with stages: goanalysis_metalinter: 25.557812ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 4 samples, avg is 71.8MB, max is 71.8MB
INFO Execution took 250.707618ms
```

**Comment by ldez** [comment]
The `asciicheck` linter produces the same error when run alone.

For the moment, I didn't find the root cause.

**Comment by ashierj** [comment]
Same issue when only enable one linter with LoadModeTypesInfo.

**Comment by siddharth178** [comment]
For me, if I run it with `govet` as suggested in above comment, I think it doesn't actually run? It can't find the issue `scopelint` sees.

Code I am trying -
```
package main

import (
	"fmt"
)

func main() {
	fmt.Println("Hello!")

	letters := []string{"a", "b", "c"}
	funcs := []func(){}
	for _, l := range letters {
		funcs = append(funcs, func(){
			fmt.Println(l)
		})
	}
	for _, f := range funcs {
		f()
	}
}
```

scopelint sees the error
```
$ golangci-lint run --verbose  --disable-all -E scopelint -E govet ./...
INFO [config_reader] Config search paths: [./ /Users/siddharthkarandikar/myprogs/scratch /Users/siddharthkarandikar/myprogs /Users/siddharthkarandikar /Users /]
INFO [lintersdb] Active 2 linters: [govet scopelint]
INFO [loader] Go packages loading at mode 575 (compiled_files|name|deps|exports_file|files|imports|types_sizes) took 190.998301ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 203.1µs
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [runner] Processors filtering stat (out/in): path_shortener: 1/1, filename_unadjuster: 1/1, autogenerated_exclude: 1/1, source_code: 1/1, max_per_file_from_linter: 1/1, skip_files: 1/1, skip_dirs: 1/1, diff: 1/1, exclude-rules: 1/1, path_prefixer: 1/1, exclude: 1/1, nolint: 1/1, uniq_by_line: 1/1, max_same_issues: 1/1, max_from_linter: 1/1, cgo: 1/1, path_prettifier: 1/1, identifier_marker: 1/1, severity-rules: 1/1, sort_results: 1/1
INFO [runner] processing took 234.9µs with stages: nolint: 52.296µs, identifier_marker: 44.074µs, path_prettifier: 32.8µs, autogenerated_exclude: 31.528µs, exclude-rules: 27.13µs, source_code: 24.317µs, skip_dirs: 9.566µs, cgo: 3.946µs, max_same_issues: 1.963µs, filename_unadjuster: 1.91µs, uniq_by_line: 1.233µs, path_shortener: 1.205µs, max_from_linter: 690ns, max_per_file_from_linter: 480ns, severity-rules: 394ns, diff: 380ns, skip_files: 359ns, exclude: 281ns, sort_results: 178ns, path_prefixer: 170ns
INFO [runner] linters took 22.078337ms with stages: goanalysis_metalinter: 21.768691ms
main.go:14:16: Using the variable on range scope `l` in function literal (scopelint)
			fmt.Println(l)
			            ^
INFO File cache stats: 1 entries of total size 252B
INFO Memory: 4 samples, avg is 71.8MB, max is 71.8MB
INFO Execution took 228.39226ms
```

but exportloopref doesn't!
```
$ golangci-lint run --verbose  --disable-all -E exportloopref -E govet ./...
INFO [config_reader] Config search paths: [./ /Users/siddharthkarandikar/myprogs/scratch /Users/siddharthkarandikar/myprogs /Users/siddharthkarandikar /Users /]
INFO [lintersdb] Active 2 linters: [exportloopref govet]
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|files|name|deps|exports_file|imports) took 185.212622ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 215.812µs
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [runner] processing took 3.727µs with stages: max_same_issues: 951ns, skip_dirs: 361ns, max_from_linter: 260ns, nolint: 225ns, cgo: 184ns, autogenerated_exclude: 169ns, uniq_by_line: 169ns, identifier_marker: 168ns, filename_unadjuster: 167ns, skip_files: 167ns, path_prettifier: 166ns, diff: 156ns, severity-rules: 95ns, exclude: 80ns, source_code: 70ns, max_per_file_from_linter: 70ns, path_shortener: 69ns, sort_results: 68ns, exclude-rules: 67ns, path_prefixer: 65ns
INFO [runner] linters took 23.1651ms with stages: goanalysis_metalinter: 23.112754ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 4 samples, avg is 72.1MB, max is 72.3MB
INFO Execution took 222.178186ms
```

**Comment by cbandy** [comment]
`scopelint` and `exportloopref` are different. Your issue is unrelated, @siddharth178.

https://github.com/kyoh86/scopelint#obsoleted

**Comment by siddharth178** [comment]
Yes, i agree. @cbandy  I thought `scopelint` features will be available in `exportloopref` and I find checks done on above code by `scopelint` important.

**Comment by ldez** [comment]
I find the root cause :tada:
I created a fix #1733
